platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/utils/StringUtils.java

```diff
@@ -0,0 +1,25 @@
+package com.mapbox.mapboxsdk.utils;
+
+import android.support.annotation.Keep;
+import android.support.annotation.NonNull;
+
+import java.text.Normalizer;
+
+/**
+ * String utility class used by core from jni.
+ */
+@Keep
+class StringUtils {
+
+  /**
+   * Normalises String input and strip diacritics from it.
+   *
+   * @return normalised String with stripped diacritics.
+   */
+  @Keep
+  @NonNull
+  static String unaccent(@NonNull String value) {
+    return Normalizer.normalize(value, Normalizer.Form.NFD)
+      .replaceAll("\\p{InCombiningDiacriticalMarks}+", "");
```

Comment:
Awesome! From https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html, it looks like you can use match any Unicode block, in which case we can get a little closer to the nunicode implementation by also matching against `Combining Diacritical Marks Extended/for Symbols/Supplement`.

Reply:
@ChrisLoer having a hard time figuring out the correct configuration for above 😅Typography is clearly not my cup of tea. Based on above, I should allow any unicode character? Thus something as`"[^\\p{ASCII}]"` or do I need to work on allowing multiple `\\p{}` items (For example: InCombiningDiacriticalMarks in combination with configurations on symbols/supplement from  https://www.regular-expressions.info/unicode.html).

Reply:
Err, I think it'll be something like:

`(\\p{InCombiningDiacriticalMarks}|\\p{InCombiningDiacriticalMarksExtended}|\\p{InCombiningDiacriticalMarksForSymbols}|\\p{InCombiningDiacriticalMarksSupplement})+`

?

Reply:
Wow, I wouldn't have figured that out on my own. Thank you, I now went with:

> (\\p{InCombiningDiacriticalMarks}|\\p{InCombiningDiacriticalMarksForSymbols}|\\p{InCombiningDiacriticalMarksSupplement})+

 as `InCombiningDiacriticalMarksExtended` didn't compile for me:

<img width="675" alt="screen shot 2018-09-11 at 10 02 23" src="https://user-images.githubusercontent.com/2151639/45346705-431f6980-b5aa-11e8-9fbb-7a21bd9a1068.png">

Reply:
Huh, I wonder why "Extended" wasn't included (although the documentation is pretty vague on the naming -- it sounds like you just strip the spaces from the Unicode block name). That will make for a small difference from the nunicode implementation, but we already tolerate differences in collator behavior from device to device and locale to locale so I think it should be OK.